linetools/isgm/abscomponent.py

```diff
@@ -0,0 +1,232 @@
+       e.g. (8,1) for OI
+    zcomp : float
+       Component redshift
+    vlim : Quantity array
+       Velocity limits of the component
+       e.g.  [-300,300]*u.km/u.s
+    A : int
+        Atomic mass -- used to distinguish isotopes
+    Ej : Quantity
+       Energy of lower level (1/cm)
+    """
+    @classmethod
+    def from_abslines(cls, abslines):
+        """Instantiate from a list of AbsLine objects
+        Parameters
+        ----------
+        abslines : list 
+          List of AbsLine objects
+        """
+        # Check
+        if not isinstance(abslines,list):
+            raise IOError('Need a list of AbsLine objects')
+        # Instantiate with the first line
+        init_line = abslines[0]
+        slf = cls( init_line.attrib['coord'],
+           (init_line.data['Z'],init_line.data['ion']),
+           init_line.attrib['z'], init_line.analy['vlim']) 
+        slf._abslines.append(init_line)
+        # Append with component checking
+        if len(abslines) > 1:
+            for absline in abslines[1:]:
+                slf.add_absline(absline)
+        # Return
+        return slf
+
+    # Initialize with wavelength
+    def __init__(self, radec, Zion, z, vlim, Ej=0./u.cm, A=None, Ntup=None):
+        """  Initiator
+
+        Parameters
+        ----------
+        radec : tuple or SkyCoord
+          (RA,DEC) in deg
+        Zion : tuple 
+          Atomic number, ion -- (int,int) 
+             e.g. (8,1) for OI
+        z : float
+          Absorption redshift
+        vlim : Quantity array
+          Velocity limits of the component
+          e.g.  [-300,300]*u.km/u.s
+        A : int, optional
+          Atomic mass -- used to distinguish isotopes
+        Ntup : tuple
+          (int,float,float)
+          (flgN,logN,sigN)
+          flgN : Flag describing N measurement
+          logN : log10 N column density
+          sigN : Error in log10 N
+        Ej : Quantity, optional
+           Energy of lower level (1/cm)
+        """
+
+        # Required
+        if isinstance(radec,(tuple)):
+            self.coord = SkyCoord(ra=radec[0], dec=radec[1])
+        elif isinstance(radec,SkyCoord):
+            self.coord = radec
+        self.Zion = Zion
+        self.zcomp = z
+        self.vlim = vlim
+
+        # Optional
+        self.A = A
+        self.Ej = Ej
+        if Ntup is not None:
+            self.flgN = Ntup[0]
+            self.logN = Ntup[1]
+            self.sigN = Ntup[2]
+
+        # Other
+        self._abslines = []
+
+    def add_absline(self,absline):
+        """Add an AbsLine object to the component if it satisfies
+        all of the rules.
+        For velocities, we demands that the new line has a velocity
+          range that is fully encompassed by the component.
+
+        Parameters
+        ----------
+        absline : AbsLine
+        """
+        # Perform easy checks
+        test = self.coord.separation(absline.attrib['coord']) < 0.1*u.arcsec
```

Comment:
Consider to have this tolerance parameter as optional. If so, my inclination would be to have it as an attribute at the AbsComponent level rather than in add_absline().